### DISCUSSION TITLE
How do I tell sqlc  to omit a field in the json if it's empty

### DISCUSSION THREAD

**Discussion by adityaladwa** [sqlc-dev/sqlc#1087] [Q&A]
Let's say I have a table

```sql
create table "todos" (
    "id" uuid primary key default uuid_generate_v4(),
    "title" varchar(255) not null,
    "description" text default null
);
```

After I run `sqlc generate` the following model will be generated:

```go
type Todo struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description sql.NullString    `json:"description"`
}
```

I would like to omit the _description_ if it's empty (as it's a nullable column). 

Now, I can use column override and use `string` type instead of `sql.NullString`. So now the generated type will be

```go
type Todo struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description string   `json:"description"`
}
```

The problem with this is that if _Description_ is `nil`, it won't be omitted from the `json` response. So the response would be something like this:

```json
{
    "id": "c0507d40-de51-451f-b84e-c2f06af7710f",
    "title": "This is a todo",
    "description": null
}

```

Is there a way I can tell `sqlc` to add `omitempty` tag to a particular field? (Description field in this case)

Any other way to solve this issue?

**Comment by kyleconroy** [comment]
> Is there a way I can tell sqlc to add omitempty tag to a particular field? (Description field in this case)

There is currently not a way a to do this. What you can do is use a custom struct and override the output of the query to return that struct.

**Reply by adityaladwa** [reply]
Can you share an example or documentation for the same?

**Reply by adityaladwa** [reply]
@kyleconroy can you share an example as I can't find any documentation on how to override output of query to return a custom struct.

**Reply by adityaladwa** [reply]
> use a custom struct and override the output of the query to return that struct.

@kyleconroy would appreciate if you can give an example or share any links on how to do this? Thank you!

**Reply by adityaladwa** [reply]
I'm going to go ahead, and assume that currently, there is no way to return a custom struct from queries. Will be re-opening this issue till I find a good resolution

**Comment by ehfeng** [comment]
In the same vein, could sql add support for golang json struct flag `,string`? JavaScript does not support parsing int64 numbers, so json marshaling as strings is preferred.

```go
type User struct {
  Id int64 `json:"id,string"`
}
```

**Comment by abdullah2993** [comment]
I have opened a [PR](https://github.com/sqlc-dev/sqlc/pull/3117)  for adding support for `omitempty`
